e2e/cypress/integration/channel/message_permalink.js

```diff
@@ -0,0 +1,101 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+// ***************************************************************
+// - [number] indicates a test step (e.g. 1. Go to a page)
+// - [*] indicates an assertion (e.g. * Check the title)
+// - Use element ID when selecting an element. Create one if none.
+// ***************************************************************
+
+/* eslint max-nested-callbacks: ["error", 7] */
+
+import users from '../../fixtures/users.json';
+import * as TIMEOUTS from '../../fixtures/timeouts';
+
+function createNewDMChannel(channelname) {
+    cy.get('#addDirectChannel').scrollIntoView().click();
+
+    cy.get('#selectItems').within(() => {
+        cy.get('input[type="text"]').scrollIntoView().type(channelname, {force: true});
+    });
+    cy.contains('.more-modal__description', channelname).click({force: true});
+    cy.get('#saveItems').click();
+}
```

Comment:
We have this same function at `search_user_post_spec.js`.  Maybe let's move this to `ui_command` as common command or just use `cy.apiCreateChannel` to create new channel.

Reply:
+1 to this, preference being leveraging the API.

Reply:
resolved